packages/component-header/package.json

```diff
@@ -37,7 +37,7 @@
     "build-storybook": "build-storybook -s ./dist"
   },
   "peerDependencies": {
-    "@asu/components-core": "^2.1.0",
+    "@asu/components-core": "^2.1.0 || ^3.0.0",
```

Comment:
Remove `components-core` as peerDep. Move referenced code to `utilities/` and reference it by path so it gets compiled into package.

Also, let's remove peerDeps for `https://github.com/ASU/asu-unity-stack/pull/1160#pullrequestreview-1704711882` and `prop-types` and `styled-components`.